Collection.UnitTests/CollectionTests.cs

```diff
@@ -69,8 +70,237 @@ public void Test_Collection_GetByInvalidIndex()
         {
             var coll = new Collection<string>("Peter", "Ivan");
 
-            Assert.That(() => { var item = coll[2]; }, 
+            Assert.That(() => { var item = coll[2]; },
                 Throws.InstanceOf<ArgumentOutOfRangeException>());
         }
+
+        [Test]
+        public void Test_Collection_SetByInvalidIndex()
+        {
+            var coll = new Collection<string>("John", "Maria", "Stacey");
+
+            Assert.That(() => { coll[100] = "Shawn"; },
+                Throws.InstanceOf<ArgumentOutOfRangeException>());
+        }
+
+        [Test]
+        public void Test_Collection_SetByNegativeInvalidIndex()
+        {
+            var coll = new Collection<string>("John", "Maria", "Stacey");
+
+            Assert.That(() => { coll[-1] = "Shawn"; },
+                Throws.InstanceOf<ArgumentOutOfRangeException>());
+        }
+
+        [Test]
+        public void Test_Collection_AddWithGrow()
+        {
+            var coll = new Collection<int>();
+            for (int i = 0; i < 19; i++)
+            {
+                coll.Add(i);
+            }
+            Assert.That(coll.ToString(), Is.EqualTo("[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]"));
+        }
+
+        [Test]
+        public void Test_Collection_AddRange()
+        {
+            var coll = new Collection<int>(4, 7, 9);
+            coll.AddRange(10, 11, 12, 13);
+
+            Assert.That(coll.ToString, Is.EqualTo("[4, 7, 9, 10, 11, 12, 13]"));
+        }
+
+        [Test]
+        public void Test_Collection_AddRangeWithGrow()
+        {
+            var coll = new Collection<int>();
+            var oldCapacity = coll.Capacity;
+            var newColl = Enumerable.Range(1000, 2000).ToArray();
+
+            coll.AddRange(newColl);
+
+            Assert.That(coll.ToString, Is.EqualTo("[" + string.Join(", ", newColl) + "]"));
+            Assert.That(coll.Capacity, Is.GreaterThanOrEqualTo(oldCapacity));
+            Assert.That(coll.Capacity, Is.GreaterThan(coll.Count));
+        }
+
+        [Test]
+        public void Test_Collection_InsertAtStart()
+        {
+            var coll = new Collection<double>(3.6, 7.8, 9.9);
+            coll.InsertAt(0, 1.1);
+
+            Assert.That(coll.ToString, Is.EqualTo("[1,1, 3,6, 7,8, 9,9]"));
+        }
+
+        [Test]
+        public void Test_Collection_InsertAtMiddle()
+        {
+            var coll = new Collection<char>('a', 'b', 'c', 'd');
+            coll.InsertAt(2, 'm');
+
+            Assert.That(coll.ToString, Is.EqualTo("[a, b, m, c, d]"));
+        }
+
+        [Test]
+        public void Test_Collection_InsertWithGrow()
```

Comment:
This is most likely for the InsertAtEnd collection test.

Reply:
Yes, that's right, I fixed the name of the method, thanks!